### PR TITLE
Update latest Scala 2 and Scala 3 versions

### PR DESCRIPTION
**Pull request by WojciechMazur**
### Description
Updates to latest minor version of Scala 2 or patch version of Scala 3: 
* 2.12.19 -> 2.12.20
* 2.13.14 -> 2.13.15
* 3.3.3    -> 3.3.4
* 3.5.0    -> 3.5.2

### CODE REVIEW

File: scripts/create_repository.py [hunk 1]
```diff
@@ -104,7 +104,9 @@ def map_to_resolved_artifacts(output) -> List[ResolvedArtifact]:
 
 def resolve_artifacts_with_checksums_and_direct_dependencies(root_artifacts) -> List[ResolvedArtifact]:
   command = f'cs resolve {' '.join(root_artifacts)}'
-  output = subprocess.run(command, capture_output=True, text=True, shell=True).stdout.splitlines()
+  proc = subprocess.run(command, capture_output=True, text=True, shell=True)
+  print(proc.stderr)
```
Comment (WojciechMazur) [inline scripts/create_repository.py:108]: I left it here, becouse if some artifact cannot be find (see kind-projector) we're left without any context

File: third_party/repositories/scala_3_3.bzl [hunk 2]
```diff
@@ -143,8 +144,8 @@ artifacts = {
         ],
     },
     "org_scala_sbt_compiler_interface": {
-        "artifact": "org.scala-sbt:compiler-interface:1.10.1",
-        "sha256": "6f9982aea22fa17fef0315abd6925940dbf57fa7033535ad2176aba240aeaa1d",
+        "artifact": "org.scala-sbt:compiler-interface:1.9.6",
+        "sha256": "b986ad1cfcef14c8abeaa3e788df28be9150acfbf587b5f7aa92c1b043c4fa8d",
```
Comment (WojciechMazur) [inline third_party/repositories/scala_3_3.bzl:148]: I don't know why the previously declared version was higher than minor version used in the compiler.

**Comment by WojciechMazur** [comment]
For reviewers: 
The first commit contains most of important changes: changes to versions and SHA's 
The next commits are fixing rest of the build to ensure usage of the latest version in tests

**Review by simuons — APPROVED** [review]

**Comment by simuons** [comment]
Thanks, @WojciechMazur!

**Pull request merged**